Chalenges/set/Set.js

```diff
@@ -0,0 +1,95 @@
+class Set {
+  constructor() {
+    this.set = {};
+  }
+  add(value) {
+    if(!this.set[value]) {
+      this.set[value] = value;
+    } else {
+      return 'this is already here try adding something else';
+    }
+  }
+  remove(value) {
+    if(this.set[value]) {
+      delete this.set[value];
+    } else {
+      return 'this was not here to begin with';
+    }
+  }
+  has(value) {
+    const objVals = Object.values(this.set);
+    const trueOrFalse = { has: false };
+    
+    for(let i = 0; i < objVals.length; i++) {
+      const comp = objVals[i];
+      if(value == comp) {
+        trueOrFalse.has = true;
+      }
+    }
+    
+    return trueOrFalse.has;
+  }
+  intersection(set) {
+    const mySet = this;
+    return Set.intersection(set, mySet);
+  }
+  union(set) {
+    const mySet = this;
+    return Set.union(set, mySet);
+  }
+  difference(set) {
+    const mySet = this;
+    return Set.difference(set, mySet);
+  }
+  static intersection(setA, setB) {
+    const aVals = Object.values(setA.set);
+    const newSet = new Set();
+
+    for(let i = 0; i < aVals.length; i++) {
+      const aVal = aVals[i];
+     
+      if(setB.set[aVal]) {
+        newSet.set[aVal] = aVal;
+      }
+    }
+    return newSet;
+  }
+  static union(setA, setB) {
```

Comment:
don't duplicate your logic. use your instance methods here or use your static methods in your instance methods